### PR TITLE
SDA-4652 | feat: add list oidc providers command

### PR DESCRIPTION
**Pull request by gdbranco**
Related issue: https://issues.redhat.com/browse/SDA-4652
# What
Adds listing of identity providers in current AWS account filtering `red-hat-managed:true` tags

For the identity providers which are created through an OIDC configuration is shows the cluster ID as empty ("") and the OIDC configuration ID is not yet retrieved. This will be part of second PR 

```
./rosa list oidc-provider                                    
I: Fetching OIDC providers
/ time=2023-05-29T09:04:19-03:00 level=warning msg=Throttling Rate limit exceeded. Retrying the request again
OIDC PROVIDER ARN                                                                                                Cluster ID
arn:aws:iam::xxx:oidc-provider/d1b0cha94ris3q.cloudfront.net/xxx           
arn:aws:iam::xxx:oidc-provider/d3gt1gce2zmg3d.cloudfront.net/xxx           
arn:aws:iam::xxx:oidc-provider/d3gt1gce2zmg3d.cloudfront.net/xxx           
arn:aws:iam::xxx:oidc-provider/d3gt1gce2zmg3d.cloudfront.net/xxx           xxx
arn:aws:iam::xxx:oidc-provider/d3gt1gce2zmg3d.cloudfront.net/xxx           xxx
arn:aws:iam::xxx:oidc-provider/d3gt1gce2zmg3d.cloudfront.net/xxx           xxx
arn:aws:iam::xxx:oidc-provider/xxx.s3.us-east-1.amazonaws.com                             
arn:aws:iam::xxx:oidc-provider/xxx.s3.us-east-1.amazonaws.com                        
arn:aws:iam::xxx:oidc-provider/xxx.s3.us-east-1.amazonaws.com                              
arn:aws:iam::xxx:oidc-provider/rh-oidc-dev.s3.us-east-1.amazonaws.com/xxx  xxx
arn:aws:iam::xxx:oidc-provider/rh-oidc-dev.s3.us-east-1.amazonaws.com/xxx  
arn:aws:iam::xxx:oidc-provider/rh-oidc-dev.s3.us-east-1.amazonaws.com/xxx  
arn:aws:iam::xxx:oidc-provider/rh-oidc-dev.s3.us-east-1.amazonaws.com/xxx  
```

### CODE REVIEW

File: cmd/list/oidcprovider/cmd.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		spin = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+	}
+	if spin != nil {
+		r.Reporter.Infof("Fetching OIDC providers")
+		spin.Start()
+	}
+
+	clusterKey := ""
+	if cmd.Flags().Changed("cluster") {
+		iclusterKey, err := ocm.GetClusterKey()
```
Comment (oriAdler) [inline cmd/list/oidcprovider/cmd.go:65]:
```suggestion
		oidcClusterKey, err := ocm.GetClusterKey()
```

File: cmd/list/oidcprovider/cmd.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		os.Exit(1)
+	}
+
+	if len(providers) == 0 {
+		r.Reporter.Infof("No OIDC providers available")
+		os.Exit(0)
+	}
+	if output.HasFlag() {
+		outList := []map[string]interface{}{}
+		for _, provider := range providers {
```
Comment (oriAdler) [inline cmd/list/oidcprovider/cmd.go:88]: Should we do the filtering inside `ListOidcProviders()` if a cluster ID was provided?

File: cmd/list/oidcprovider/cmd.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+			os.Exit(1)
+		}
+		os.Exit(0)
+	}
+
+	// Create the writer that will be used to print the tabulated results:
+	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
+	fmt.Fprintf(writer, "OIDC PROVIDER ARN\tCluster ID\n")
+	for _, oidcProvider := range providers {
+		if clusterKey != "" && oidcProvider.ClusterId != clusterKey {
```
Comment (oriAdler) [inline cmd/list/oidcprovider/cmd.go:106]: Since we already filtered out the providers that don't match the cluster ID, do we need this check here?

**Review by oriAdler — COMMENTED** [review]
Left a couple of comments, other than that LGTM.
Can you please attach an output example of the command? to see what it looks like:)

File: cmd/list/oidcprovider/cmd.go (hunk 1 above)
Comment (gdbranco) [reply cmd/list/oidcprovider/cmd.go:65]: Oh, by `i` I meant internal. Maybe I should just add a `_` instead

File: cmd/list/oidcprovider/cmd.go (hunk 3 above)
Comment (gdbranco) [reply cmd/list/oidcprovider/cmd.go:106]: We didn't here, that only got filtered if it has output flag. But anyways the other comment of filtering it in the listing itself is probably better

**Pull request merged**